### PR TITLE
refactor: implement Glossary component from Hew

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Glossary entirely replaces InfoBox and the `renderRow`/`renderResource` function in the CheckpointModal. Additionally, it is now used as the display logic for Json (now `JsonGlossary`).

## Test Plan
Check the following locations to make sure nothing looks weird:

- CheckpointModal
- ResourcePoolCard
- Resource pool config page
- ExperimentDetails FoldableHeader
- EditableMetadata (when not editing, with at least 2 key/value pairs)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
Can you add before/after screenshots to your PR description?

**Comment by mapmeld** [comment]
The JsonGlossary component used for resource pools on `/det/clusters` -- I'm not sure if this is custom CSS, or regularizing the component, but the values were previously right-aligned

Screen Shot 2023-11-09 at 3 51 18 PM

File: webui/react/src/components/JsonGlossary.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    let textValue = '';
+    if (isObject(jsonValue)) {
+      textValue = JSON.stringify(jsonValue, null, 2);
+    } else if (jsonValue === '' || jsonValue === null || jsonValue === undefined) {
+      textValue = 'N/A';
+    } else {
+      textValue = jsonValue.toString();
+    }
+    return {
+      label: typeof label === 'string' && translateLabel ? translateLabel(label) : label,
```
Comment (mapmeld) [inline webui/react/src/components/JsonGlossary.tsx:25]: nit: because `isObject` is used in the file, I would suggest import `isString` from there too

**Review by mapmeld — APPROVED** [review]
I rebased and the alignment changes are great.

Screen Shot 2023-11-13 at 3 57 58 PM

One thing - the vertical spacing in this card between rows is unexpectedly high, but I don't know is that consistent with our design or one label carrying over to the next line?

**Comment by EmilyBonar** [comment]
Ah you're right. It looks like that came from using the `Body` component. I'll go ahead and fix that.

**Pull request merged**